File: java/leetcode/src/main/java/com/leetcode/tree/Binary_Tree_Level_Order_Traversal_102.java
package com.leetcode.tree;


import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class Binary_Tree_Level_Order_Traversal_102 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> lists = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        if (root !=null) {
            queue.offer(root);
        }
        while (!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> list = new ArrayList<>();
            for (int i = 0; i < size; i++) {
                TreeNode node = queue.poll();
                list.add(node.val);
                if (node.left != null) {
                    queue.offer(node.left);
                }
                if (node.right != null) {
                    queue.offer(node.right);
                }
            }
            lists.add(list);
        }
        return lists;
    }

}

File: js/algrithm/tree/Binary_Tree_Level_Order_Traversal_102.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    const ans = []
    const queue = []
    if (!root) return ans
    queue.push(root)
    var level = 0
    while (queue.length) {
        const size = queue.length
        ans[level] = []
        for (let i = 0; i < size; i++) {
            const cur = queue.shift()
            ans[level].push(cur.val)
            if (cur.left) {
                queue.push(cur.left)
            } 
            if (cur.right) {
                queue.push(cur.right)
            }
        }
        level++
    }
    return ans
};